File: Whatsapp/OpenWA/sdk/java/src/main/java/com/rmyndharis/openwa/resources/SearchResource.java
package com.rmyndharis.openwa.resources;

import com.rmyndharis.openwa.OpenWAClient;
import com.rmyndharis.openwa.http.HttpMethod;
import com.rmyndharis.openwa.model.SearchQuery;
import com.rmyndharis.openwa.model.SearchResults;

/**
 * Search resource — full-text message search across sessions via {@code GET /search}.
 *
 * <p>Requires an OPERATOR-level API key. {@code q} is the only required parameter.
 */
public final class SearchResource {
    private final OpenWAClient client;

    public SearchResource(OpenWAClient client) {
        this.client = client;
    }

    /**
     * Search persisted messages via the active search provider.
     *
     * @param params query parameters; {@code q} must be non-null and non-blank.
     * @throws IllegalArgumentException if {@code params.q} is null or blank.
     */
    public SearchResults search(SearchQuery params) {
        if (params == null) {
            throw new IllegalArgumentException("Search params must not be null.");
        }
        if (params.q() == null || params.q().isBlank()) {
            throw new IllegalArgumentException("Search parameter \"q\" is required and must be non-empty.");
        }
        return client.request(HttpMethod.GET, "/api/search", params, null, SearchResults.class);
    }
}

File: Whatsapp/OpenWA/sdk/php/src/Resources/SearchResource.php
<?php

declare(strict_types=1);

namespace OpenWA\Resources;

use OpenWA\Http\HttpExecutor;

/**
 * Search resource — full-text search across persisted messages.
 *
 * Backed by src/modules/search/search.controller.ts (GET /search). Requires an
 * active search provider (SEARCH_PROVIDER); responds 501 when none is configured
 * and 400 for an empty/whitespace `q`.
 */
class SearchResource
{
    private HttpExecutor $http;

    public function __construct(HttpExecutor $http)
    {
        $this->http = $http;
    }

    /**
     * Search persisted messages via the active search provider.
     *
     * Only `q` is required; every other filter is optional and null values are
     * dropped before the request is sent (so absent optionals are not serialized
     * into the query string). Numeric filters are passed through as-is; the
     * gateway coerces and validates them server-side.
     *
     * @param array{
     *     q:string,
     *     sessionId?:string|null,
     *     chatId?:string|null,
     *     direction?:string|null,
     *     type?:string|null,
     *     from?:string|null,
     *     dateFrom?:int|float|string|null,
     *     dateTo?:int|float|string|null,
     *     limit?:int|string|null,
     *     offset?:int|string|null
     * } $params Query filters. `q` must be non-empty.
     *
     * @return array{
     *     hits:array<int,array<string,mixed>>,
     *     total:int,
     *     tookMs:int,
     *     provider:string
     * } Decoded SearchResults payload (empty array only on an unexpected null body).
     */
    public function search(array $params): array
    {
        return $this->http->request('GET', '/api/search', $params) ?? [];
    }
}
